nupic/algorithms/anomaly.py

```diff
@@ -167,10 +178,35 @@ def __str__(self):
   def __setstate__(self, state):
     """deserialization"""
     self.__dict__.update(state)
-
     if not hasattr(self, '_mode'):
       self._mode = Anomaly.MODE_PURE
     if not hasattr(self, '_movingAverage'):
       self._movingAverage = None
     if not hasattr(self, '_binaryThreshold'):
       self._binaryThreshold = None
+
+
+  def getComputeFn(self, mode):
+    """
+    assing compute() function to be used; 
+    all these functions must take these arguments:
+     activeColumns, prevPredictedColumns, inputValue=None, timestamp=None
+    and return anomaly score
+
+    @param mode   if string: we assign from Anomaly.MODE_* defined functions
+                  if func: we will use that function to compute anomaly score
+    @return a function that is used for anomaly computation
+    """
+    if mode == Anomaly.MODE_PURE:
+      return self.computeRaw
+    elif mode == Anomaly.MODE_LIKELIHOOD:
+      self._likelihood = AnomalyLikelihood() # probabilistic anomaly
+      return self.computeLikelihood
+    elif mode == Anomaly.MODE_WEIGHTED:
+      self._likelihood = AnomalyLikelihood() # probabilistic anomaly
+      return self.computeWeighted
+    elif isinstance(mode, FunctionType):
+      print "Anomaly: using custom compute() function"
+      return mode
+    else:
+      raise ValueError("Anomaly: computeFn has to be one of '%s' but is '%s' " % (Anomaly._supportedModes, mode) )
```

Comment:
The exception format text should be `"Anomaly: mode has to be a callable or one of '%s' but is '%s' "`